smac/intensification/intensification.py

```diff
@@ -112,13 +110,34 @@ def __init__(self, tae_runner: ExecuteTARun,
         self.use_ta_time_bound = use_ta_time_bound
         self.min_chall = min_chall
 
-    def intensify(self, challengers: typing.Optional[typing.List[Configuration]],
-                  optimizer: typing.Optional['SMBO'],
-                  incumbent: Configuration,
-                  run_history: RunHistory,
-                  aggregate_func: typing.Callable,
-                  time_bound: float = float(MAXINT),
-                  log_traj: bool = True) -> typing.Tuple[Configuration, float]:
+        # stage variables
+        # the intensification procedure is divided into 3 'stages':
+        # 1. add incumbent run
+        # 2. race challenger
+        # 3. race against configuration for a new incumbent
+        self.run_incumbent = True
+        self.run_challenger = False
+        self.first_run = True
+        self.n_iters = 0
+
+        # challenger related variables
+        self._chall_indx = 0
+        self.running_challenger = None
```

Comment:
I guess this variable should be named `current_challenger` as it does not specifiy whether something is running, but what is running.